lessons/02-javascript/README.md

```diff
@@ -152,6 +157,8 @@ for (var key in object) {
 }
 ```
 
+Note that the order of iteration is not guarenteed with this syntax and later, if you use libraries which modify the prototype (more on this later) of base types, can have strange behavior.
```

Comment:
`hasOwnProperty` is preferred in most cases because it doesn't search through prototype keys.
For example if you have an object which extends a function prototype, then using `hasOwnProperty` will only look through that instance's keys. Good for ignoring internal properties and methods.

From the docs
> Every object descended from Object inherits the hasOwnProperty method. This method can be used to determine whether an object has the specified property as a direct property of that object; unlike the in operator, this method does not check down the object's prototype chain.